src/components/policy/policy_external/src/policy_table/validation.cc

```diff
@@ -206,6 +206,10 @@ bool UsageAndErrorCounts::Validate() const {
   return true;
 }
 bool ConsentRecords::Validate() const {
+  if (PT_SNAPSHOT != GetPolicyTableType()) {
+    return !external_consent_status_groups->is_initialized();
+  }
+
```

Comment:
@LuxoftAKutsan  According to requirements external_consent_status_groups is allowed to be present only in snapshots. The code already has contained required validation, see ConsentRecords::is_valid.